src/fides/api/ops/api/v1/endpoints/privacy_notice_endpoints.py

```diff
@@ -0,0 +1,218 @@
+from typing import Dict, List, Optional, Tuple
+
+from fastapi import Depends, Security
+from fastapi_pagination import Page, Params
+from fastapi_pagination.bases import AbstractPage
+from fastapi_pagination.ext.sqlalchemy import paginate
+from loguru import logger
+from pydantic import conlist
+from sqlalchemy.orm import Session
+from starlette.exceptions import HTTPException
+from starlette.status import (
+    HTTP_200_OK,
+    HTTP_404_NOT_FOUND,
+    HTTP_422_UNPROCESSABLE_ENTITY,
+)
+
+from fides.api.ctl.sql_models import System  # type: ignore
+from fides.api.ops.api import deps
+from fides.api.ops.api.v1 import scope_registry
+from fides.api.ops.api.v1 import urn_registry as urls
+from fides.api.ops.common_exceptions import ValidationError
+from fides.api.ops.models.privacy_notice import (
+    PrivacyNotice,
+    PrivacyNoticeRegion,
+    check_conflicting_data_uses,
+)
+from fides.api.ops.schemas import privacy_notice as schemas
+from fides.api.ops.util.api_router import APIRouter
+from fides.api.ops.util.oauth_util import verify_oauth_client
+
+router = APIRouter(tags=["Privacy Notice"], prefix=urls.V1_URL_PREFIX)
+
+
+@router.get(
+    urls.PRIVACY_NOTICE,
+    status_code=HTTP_200_OK,
+    response_model=Page[schemas.PrivacyNoticeResponse],
+    dependencies=[
+        Security(verify_oauth_client, scopes=[scope_registry.PRIVACY_NOTICE_READ])
+    ],
+)
+def get_privacy_notice_list(
+    *,
+    db: Session = Depends(deps.get_db),
+    params: Params = Depends(),
+    show_disabled: Optional[bool] = True,
+    region: Optional[PrivacyNoticeRegion] = None,
+    systems_applicable: Optional[bool] = True,
+) -> AbstractPage[PrivacyNotice]:
+    """
+    Return a paginated list of PrivacyNotice records in this system.
+    Includes some query params to help filter the list if needed
+    """
+    logger.info("Finding all PrivacyNotices with pagination params '{}'", params)
+    notice_query = db.query(PrivacyNotice)
+    if not show_disabled:
+        notice_query = notice_query.filter(PrivacyNotice.disabled.is_(False))
+    if region is not None:
+        notice_query = notice_query.filter(PrivacyNotice.regions.contains([region]))
+    if systems_applicable:
+        data_uses = System.get_system_data_uses(db)
```

Comment:
i didn't love having to execute 2 queries here, but the join query here was becoming quite complex for me, specifically because the data uses are string fields within within a JSON blob column (that's actually an array) on the `System` table.

rather than continue down that rabbit hole, this was a lot simpler for implementation, and we're still issuing a constant (2) number of queries (not N), so i felt this was fine.

Reply:
yeah, i was very happy when i found overlap - it's a nice little function!